### PR TITLE
Faq front

### PR DESCRIPTION
**Pull request by E-zh**

### CODE REVIEW

File: src/Presenters/FaqPresenter.php [hunk 1]
```diff
@@ -150,4 +150,25 @@ public function deleteQuestionAnswer($id)
 
         return true;
     }
+
+    /**
+     * Get all active categories of FAQ
+     *
+     * @return Collection|Category[]
+     */
+    public function getAllActiveCategories()
```
Comment (garf) [inline src/Presenters/FaqPresenter.php]: `getActiveCategories()`

File: src/Presenters/FaqPresenter.php [hunk 2]
```diff
@@ -150,4 +150,25 @@ public function deleteQuestionAnswer($id)
...
+    {
+        return Category::where('status', Category::STATUS_ACTIVE)->paginate(FaqPresenter::PER_PAGE);
+    }
+
+    /**
+     * Get question-answer
+     *
+     * @param string $slug
+     * @return object
```
Comment (garf) [inline src/Presenters/FaqPresenter.php]: `QuestionAnswer`?

File: src/resources/views/site/index.blade.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    <div class="container">
+        <div class="panel panel-default">
+            <div class="panel-heading">
+                <h3>{{ trans('faq::faq.front.index-head') }}</h3>
+            </div>
+            <div class="panel-body">
+                @if(!$categories->isEmpty())
+                    <div class="panel-group" id="accordion" role="tablist" aria-multiselectable="true">
+                        @foreach($categories as $category)
+                            @if(!$category->questionsAnswers->isEmpty())
```
Comment (garf) [inline src/resources/views/site/index.blade.php]: I would put the part inside `foreach` into separate view. This view becomes too complicated.

**Review by garf — APPROVED** [review]
Small comments and can be merged.

File: src/Presenters/FaqPresenter.php (hunk 2 above)
Comment (E-zh) [reply src/Presenters/FaqPresenter.php]: Yes) I can rename, but don't know what name to choose)))

Comment (garf) [reply src/Presenters/FaqPresenter.php]: Я имел ввиду в аннотации укажи тип. :))

File: src/Presenters/FaqPresenter.php [hunk 4]
```diff
@@ -150,4 +150,25 @@ public function deleteQuestionAnswer($id)
...
+     */
+    public function getBySlugQuestionAnswer($slug)
```
Comment (garf) [inline src/Presenters/FaqPresenter.php]: getQuestionAnswerBySlug

File: src/resources/views/site/index-qa.blade.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
+@if(!$category->questionsAnswers->isEmpty())
```
Comment (garf) [inline src/resources/views/site/index-qa.blade.php]: Let's name this file _qa-card.blade.php

Comment (garf) [reply src/resources/views/site/index-qa.blade.php]: No. _qa-category.blade.php. Ignore previous comment.

File: src/Presenters/FaqPresenter.php (hunk 2 above)
Comment (E-zh) [reply src/Presenters/FaqPresenter.php]: Так указан же в строке 167?

Comment (garf) [reply src/Presenters/FaqPresenter.php]: А здесь в аннотации? =) Должно же быть не object а QuestionAnswer|null =)

Comment (E-zh) [reply src/Presenters/FaqPresenter.php]: Ааааа) Исправлено :)

**Review by garf — APPROVED** [review]

**Pull request merged**